sdk/core/azure-core-amqp/src/impl/rust_amqp/rust_amqp/rust_wrapper/src/rust_error.rs

```diff
@@ -7,10 +7,11 @@ cspell: words reqwest repr staticlib dylib brotli gzip
 
 use std::ffi::c_char;
 
-pub struct RustError(azure_core::Error);
+#[derive(Debug)]
+pub struct RustError(Box<dyn std::error::Error + Send + Sync>);
 
 impl RustError {
-    pub fn new(error: azure_core::Error) -> Self {
+    pub fn new(error: Box<dyn std::error::Error + Send + Sync>) -> Self {
```

Comment:
Should I assume this isn't possible to use in wasm32 anyway? `Send + Sync` won't be available there.

Reply:
You should `cargo build --target wasm32-unknown-unknown` just to be sure, if the CI didn't catch this already.

Reply:
This is in a C++ interop layer, wasm isn't an option regardless. But I don't think send+sync is relevant here.

Reply:
C++ can be compiled to WASM as well. There's no real "interop" here akin to .NET or other languages that rely on `GetProcAddress()`. Rust compiles to LLVM and can be linked with C++ compiled to MSVC (I believe) or LLVM.

That said, I'm not hell bent on requiring this for C++, but something to keep in mind. Still, if there's a similar change to our rust repo, you should make sure to condition as appropriate. If separate methods, the wasm32 method could just be `unimplemented!()`.